### ISSUE TITLE
Disable no-default-export rule for vue files

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [palantir/tslint#4646]
### Bug Report

Filing on behalf of @akoidan.

- __TSLint version__: 5.15.0
- __TypeScript version__: 3.4.0
- __Running TSLint via__: CLI

#### ~TypeScript~ Vue code being linted

```vue
<template>
<div>asd</div>
</template>
<script lang="ts">
export default {
}
</script>
```

with `tslint.json` configuration:

```json
{
    "rules": {
        "no-default-export": true
    }
}
```

#### Actual behavior

Default export is complained on for `.vue` file, despite Vue enforcing its view components using the default export.

#### Expected behavior

Quoting https://github.com/palantir/tslint/pull/4643/#issuecomment-480595130:

There's no way to enable `no-default-export` option in `.ts` files but having them disabled in `.vue` files. [It's impossible to get rid of default export in `.vue`](https://github.com/vuejs/vue-loader/issues/1234#issuecomment-409020334).

There should be a way to enable `no-default-export` for all files _except_ `*.vue`.

**Comment by JoshuaKGoldberg** [comment]
This _would_ be a duplicate of #3447 for adding in a general `overrides` option, except that a config-wide overrides option isn't being accepted anymore. 

Thoughts @adidahiya?

**Comment by adidahiya** [comment]
I would be open to making a special exception for `.vue` files. The rule could check the file extension and bail out early (never report errors) if it is a vue file. This is less performant than conditional lint configuration since you have to visit all the .vue files, but I think it would be a reasonable compromise for the desired behavior at this point.

**Comment by akoidan** [comment]
@adidahiya well, I thought that in the first place, but I guess hard coding constants like .vue is not the best practice, there could be some drawbacks, e.g.:
 - what if vue changes the extension in the future?
 - what if someone uses custom extension for vue templates, idk like vuex
 - what if there're another single file component, probably some similar thing for angular somewhere out there.
 - what if enabling trimming space by default without user awareness could lead to some issues we are not aware yet.

That's just a few things that came in mind.

**Comment by adidahiya** [comment]
There are a few places where we hardcode constants like that, I think it's ok in rare cases.

> what if vue changes the extension in the future?

That's ok, we can update the rule if it's still a relevant concern at that future point.

> what if someone uses custom extension for vue templates, idk like vuex

What other file extensions does vue support? Do you have any examples from real projects?

> what if there're another single file component, probably some similar thing for angular somewhere out there.

I haven't seen an issue filed for that yet...

---

In general, I'd like to avoid going down the path of adding rule configuration which controls which files get linted. It's a new configuration paradigm and makes migrating configuration to ESLint in the future more complicated.

**Comment by JoshuaKGoldberg** [comment]
Per #4379, a Vue-specific workaround isn't going to be able to land in TSLint core:

> Given the lack of discussion or concrete proposal here and the upcoming deprecation of TSLint in favor of ESLint (#4534), I'm going to go ahead and close this issue and related Vue-specific reports for housekeeping purposes. 
> 
> If this is still an issue for you in [typescript-eslint](https://typescript-eslint), I'd recommend filing an issue there. Best of luck!
